### PR TITLE
DT-6455 Navigator introduction

### PR DESCRIPTION
**Pull request by partisaani**
## Proposed Changes
 - Adds an introductory modal that's shown to user prior to their first journey using navigator

## Pull Request Check List

- [x] A reasonable set of unit tests is included
- [x] Console does not show new warnings/errors
- [x] Changes are documented or they are self explanatory
- [x] This pull request does not have any merge conflicts
- [x] All existing tests pass in CI build

## Review

 - Read and verify the code changes (I suggest you review commit-by-commit for additional context of specific changes)
 - Test the functionality by running the UI locally with all popular browsers available in your platform
 - Check that the implementation matches the design, when such one is defined in a Jira issue
 - When merging to v3, I suggest it's done via a squash and merge

### CODE REVIEW

File: app/component/itinerary/ItineraryPage.js [hunk 1]
```diff
@@ -1096,14 +1111,22 @@ export default function ItineraryPage(props, context) {
...
-        <NaviContainer
-          itinerary={combinedEdges[selectedIndex]?.node}
-          focusToLeg={focusToLeg}
-          relayEnvironment={props.relayEnvironment}
-          combinedEdges={combinedEdges}
-          setNavigation={setNavigation}
-          mapRef={mwtRef.current}
-        />
+        <>
+          <NavigatorIntroModal
```
Comment (vesameskanen) [inline app/component/itinerary/ItineraryPage.js]: I wonder if rendering should be conditional: {!isNavigatorIntroDismissed && (<NavigatorIntroModal ...
Now rendering loads some resoureces even when they are not shown. IsOpen prop is then unnecessary.

File: app/translations.js [hunk 2]
```diff
@@ -2530,6 +2540,16 @@ const translations = {
     navigate: 'Navigoi',
     'navigation-description': 'Löydä perille ohjatusti',
     'navigation-header': 'Matkan seuranta',
+    'navigation-intro': 'TODO_navigation-intro_EN',
```
Comment (vesameskanen) [inline app/translations.js]: Will this become used in design plans? No references at this point.

File: app/component/itinerary/NavigatorIntro/NavigatorIntro.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  return (
+    <div className="navigator-intro-modal-content">
+      <div className="body">
+        {logo && <img src={logo} alt="navigator logo" />}
+        <FormattedMessage tagName="h2" id="navigation-intro-header" />
+        <div className="navigation-intro-body">
+          <NavigatorIntroFeature
+            icon="icon-icon_future-route"
+            iconColor={primaryColor}
+            iconBackgroundColor={config.colors?.backgroundInfo || 'transparent'}
```
Comment (vesameskanen) [inline app/component/itinerary/NavigatorIntro/NavigatorIntro.js]: Maybe `|| tranparent` is not needed. Base config sets the color.

**Review by vesameskanen — COMMENTED** [review]
Good work!

File: app/translations.js (hunk 2 above)
Comment (partisaani) [reply app/translations.js]: True. Must be an oversight leftover. I'll remove this key.

File: app/component/itinerary/ItineraryPage.js (hunk 1 above)
Comment (partisaani) [reply app/component/itinerary/ItineraryPage.js]: Good suggestion. I'll double check to make sure that it wasn't the modal that enforced this sort of implementation and then refactor if need be.

File: app/component/itinerary/NavigatorIntro/NavigatorIntro.js (hunk 3 above)
Comment (partisaani) [reply app/component/itinerary/NavigatorIntro/NavigatorIntro.js]: Will fix.

File: app/component/itinerary/ItineraryPage.js [hunk 4]
```diff
@@ -1096,14 +1111,24 @@ export default function ItineraryPage(props, context) {
...
-          setNavigation={setNavigation}
-          mapRef={mwtRef.current}
-        />
+        <>
+          {!isNavigatorIntroDismissed && (
+            <NavigatorIntroModal
+              isOpen
+              onPrimaryClick={toggleNavigatorIntro}
+              onClose={cancelNavigatorUsage}
+              isHSL={config.CONFIG !== 'hsl'}
```
Comment (vesameskanen) [inline app/component/itinerary/ItineraryPage.js]: Where did this come from? Use of word 'HSL' should be avoided in source code.

File: app/component/itinerary/NavigatorIntro/NavigatorIntroModal.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+        logo={logo}
+        onPrimaryClick={onPrimaryClick}
+        onClose={onClose}
+      />
+    </Modal>
+  );
+};
+
+NavigatorIntroModal.propTypes = {
+  isOpen: PropTypes.bool.isRequired,
```
Comment (vesameskanen) [inline app/component/itinerary/NavigatorIntro/NavigatorIntroModal.js]: isOpen prop is not needed.  Pass true to hslfi modal every time,.

**Review by vesameskanen — CHANGES_REQUESTED** [review]

File: app/component/itinerary/ItineraryPage.js (hunk 4 above)
Comment (partisaani) [reply app/component/itinerary/ItineraryPage.js]: Looks like a remnant from the time `config.navigationLogo` property was not yet being used. Will remove.

File: app/component/itinerary/NavigatorIntro/NavigatorIntroModal.js (hunk 5 above)
Comment (partisaani) [reply app/component/itinerary/NavigatorIntro/NavigatorIntroModal.js]: Streamlined in https://github.com/HSLdevcom/digitransit-ui/pull/5119/commits/dfda93fa56b1460bcdd1d63a6a4f388295d64f76

**Pull request merged**